packages/framework/src/Component/Elasticsearch/IndexFacade.php

```diff
@@ -108,20 +95,9 @@ public function export(AbstractIndex $index, IndexDefinition $indexDefinition, O
             $indexDefinition->getDomainId()
         ));
 
-        $this->sqlLoggerFacade->temporarilyDisableLogging();
-
-        $indexAlias = $indexDefinition->getIndexAlias();
+        $this->createForIndexWhenNeeded($indexDefinition, $output);
 
-        try {
-            $this->indexRepository->findCurrentIndexNameForAlias($indexAlias);
-        } catch (ElasticsearchNoAliasException $exception) {
-            $output->writeln(sprintf(
-                'Index "%s" does not exist on domain "%s"',
-                $indexDefinition->getIndexName(),
-                $indexDefinition->getDomainId()
-            ));
-            $this->create($indexDefinition, $output);
-        }
+        $this->sqlLoggerFacade->temporarilyDisableLogging();
```

Comment:
> You have moved this disable after index create - is there any reason for that?

I have not.

---

> I am not sure if it is good to create elasticsearch indexes during data export

We discussed it with @grossmannmartin to avoid getting in wrong state when you export data withoud created structure. If you want more details you can contact me in person to save our time.

Reply:
1. Ok, I was looking on method above... Anyway the logger is disabled olny due to exporting data which is fine so far.

2. This is the same as you mentioned in https://github.com/shopsys/shopsys/pull/2233#discussion_r576650595 - please continue there to resolve this thread.